osu.Game/Screens/Play/PlayerInputManager.cs

```diff
@@ -54,5 +84,38 @@ protected override void TransformState(InputState state)
                     mouse.ButtonStates.Find(s => s.Button == MouseButton.Right).State = true;
             }
         }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (parentClock == null) return;
+
+            clock.Rate = parentClock.Rate;
+            clock.IsRunning = parentClock.IsRunning;
+
+            //if a replayHandler is not attached, we should just pass-through.
+            if (UseParentState || replayInputHandler == null)

```

Comment:
I have it this way for potential future expansion (ie. the autoplay toggle in the editor) where UseParentState may be changed externally.